src/server/routes/api/v1/account.ts

```diff
@@ -0,0 +1,228 @@
+                ServeError(res, 400, "account with this username already exists")
+                return
+            }
+
+            if (req.body.username.length < 3 || req.body.username.length > 20) {
+                ServeError(res, 400, "username must be over or equal to 3 characters or under or equal to 20 characters in length")
+                return
+            }
+
+            if (
+                (
+                    req.body.username.match(/[A-Za-z0-9_\-\.]+/)
+                    ||
+                    []
+                )[0] != req.body.username
+            ) {
+                ServeError(res, 400, "username contains invalid characters")
+                return
+            }
+
+            if (req.body.password.length < 8) {
+                ServeError(res, 400, "password must be 8 characters or longer")
+                return
+            }
+
+            Accounts.create(
+                req.body.username,
+                req.body.password
+            ).then((Account) => {
+                res.cookie("auth", Authentication.create(
+                    Account, // account id
+                    (3 * 24 * 60 * 60 * 1000) // expiration time
+                ))
+                res.status(200)
+                res.end()
+            })
+            .catch(() => {
+                ServeError(res, 500, "internal server error")
+            })
+        }
+    )
+
+    router.post(
+        "/logout",
+        (req, res) => {
+            if (!Authentication.validate(req.cookies.auth)) {
+                ServeError(res, 401, "not logged in")
+                return
+            }
+
+            Authentication.invalidate(req.cookies.auth)
+            res.send("logged out")
+        }
+    )
+
+    router.patch(
+        "/dfv",
+        requiresAccount, requiresPermissions("manage"),
+        (req, res) => {
+            const Account = res.locals.acc as Accounts.Account
+
+            if (['public', 'private', 'anonymous'].includes(req.body.defaultFileVisibility)) {
+                Account.defaultFileVisibility = req.body.defaultFileVisibility
+                
+                Accounts.save()
+                
+                res.send(`dfv has been set to ${Account.defaultFileVisibility}`)
+            } else {
+                ServeError(res, 400, "invalid dfv")
+            }
+        }
+    )
+
+    router.delete("/me/:deleteFiles",
+        requiresAccount,
+        noAPIAccess,
+        parser,
+        (req, res) => {
+            const Account = res.locals.acc as Accounts.Account
+
+            const accountId = Account.id
+
+            Authentication.AuthTokens.filter(e => e.account == accountId).forEach((token) => {
+                Authentication.invalidate(token.token)
+            })
+
+            const deleteAccount = () => Accounts.deleteAccount(accountId).then(_ => res.send("account deleted"))
+
+            if (Boolean(req.params.deleteFiles)) {
```

Comment:
<img src="https://github.com/nbitzz/monofile/assets/29169102/98996a0e-fbae-4acc-a536-c619085277a2" height="16" alt=":plague:">

```suggestion
            if (req.params.deleteFiles) {
```